codec/decoder/plus/src/welsDecoderExt.cpp

```diff
@@ -234,19 +234,19 @@ void CWelsDecoder::UninitDecoder (void) {
 
   WelsEndDecoder (m_pDecContext);
 
-  if (m_pDecContext->pMemAlign != NULL) {
-    WelsLog (&m_pWelsTrace->m_sLogCtx, WELS_LOG_INFO,
-             "CWelsDecoder::UninitDecoder(), verify memory usage (%d bytes) after free..",
-             m_pDecContext->pMemAlign->WelsGetMemoryUsage());
-    delete m_pDecContext->pMemAlign;
-    m_pDecContext->pMemAlign = NULL;
-  }
+  CMemoryAlign * pSMemAllocator = m_pDecContext->pMemAlign;
 
   if (NULL != m_pDecContext) {
-    WelsFree (m_pDecContext, "m_pDecContext");
-
+    pSMemAllocator->WelsFree (m_pDecContext, "m_pDecContext");
     m_pDecContext = NULL;
   }
+
+  if (NULL != pSMemAllocator) {
+    WelsLog (&m_pWelsTrace->m_sLogCtx, WELS_LOG_INFO,
+             "CWelsDecoder::UninitDecoder(), verify memory usage (%d bytes) after free..",
+             pSMemAllocator->WelsGetMemoryUsage());
+    delete pSMemAllocator;
+  }
```

Comment:
You should set m_pDecContext->pMemAlign back to NULL here, as UninitDecoder() can be called multiple times potentially (if InitDecoder fails for some reason). You therefore also can't assume that pSMemAllocator is non-NULL on entry.